src/ui/flutter_app/lib/shared/painters/board_painter.dart

```diff
@@ -22,60 +22,38 @@ class BoardPainter extends PiecesBasePainter {
   BoardPainter({required double width}) : super(width: width);
 
   @override
-  void paint(Canvas canvas, Size size) {
-    doPaint(
-      canvas,
-      thePaint,
-      gridWidth,
-      squareWidth,
-      offsetX: AppTheme.boardPadding + squareWidth / 2,
-      offsetY: AppTheme.boardPadding + squareWidth / 2,
-    );
-  }
+  void paint(Canvas canvas, Size size) => _doPaint(canvas, thePaint);
 
   @override
-  bool shouldRepaint(CustomPainter oldDelegate) {
-    return false;
-  }
-
-  static void doPaint(
-    Canvas canvas,
-    Paint paint,
-    double gridWidth,
-    double squareWidth, {
-    required double offsetX,
-    required double offsetY,
-  }) {
-    paint.color = LocalDatabaseService.colorSettings.boardLineColor;
-    paint.style = PaintingStyle.stroke;
+  bool shouldRepaint(CustomPainter oldDelegate) => false;
 
-    final left = offsetX;
-    final top = offsetY;
+  void _doPaint(Canvas canvas, Paint paint) {
+    final offset = Offset(
+      AppTheme.boardPadding + squareWidth / 2,
+      AppTheme.boardPadding + squareWidth / 2,
+    );
 
     paint.strokeWidth = LocalDatabaseService.display.boardBorderLineWidth;
+    paint.color = LocalDatabaseService.colorSettings.boardLineColor;
+    paint.style = PaintingStyle.stroke;
 
-    if (LocalDatabaseService.display.isPieceCountInHandShown) {
-      var pieceInHandCount =
-          gameInstance.position.pieceInHandCount[PieceColor.black];
-
+    if (LocalDatabaseService.display.isPieceCountInHandShown &&
+        gameInstance.position.phase == Phase.placing) {
+      final int pieceInHandCountStr;
       if (gameInstance.position.pieceOnBoardCount[PieceColor.white] == 0 &&
           gameInstance.position.pieceOnBoardCount[PieceColor.black] == 0) {
-        pieceInHandCount = LocalDatabaseService.rules.piecesCount;
-      }
-
-      var pieceInHandCountStr = "";
-
-      // TODO: [Leptopoda] only paint it when the phase is placing.
```

Comment:
I originally planned to display the number of pieces remaining on the board when the moving phase. I didn't decide whether do it or not. I deal with the placing phase first. I need to seek the opinions of users before making a decision. So this sequence of judgments was created.

Reply:
The condition on what we display it can still be altered in the future. I just cleaned it up for now bringing quite a performance improvement.

Reply:
> > Code comments can be arbitrary, this file is exposed to the translator, so it needs to be more formal.
> 
> I don't see what exactly isn't formal in there

the current Sanmill version 

`The current Sanmill version.` is better.